src/shared/codelens/csharpCodeLensProvider.ts

```diff
@@ -61,24 +83,80 @@ export async function initialize({
  * The command that is run when user clicks on Run Local or Debug Local CodeLens
  * @param commandName - Name of the VS Code Command currently running
  * @param lambdaLocalInvokeParams - Information about the Lambda Handler to invoke locally
+ * @param processInvoker - SAM CLI Process invoker
+ * @param taskInvoker - SAM CLI Task invoker
  */
 async function onLocalInvokeCommand({
+    configuration,
+    toolkitOutputChannel,
     commandName,
     lambdaLocalInvokeParams,
+    processInvoker,
+    taskInvoker,
+    telemetryService
 }: {
+    configuration: SettingsConfiguration
+    toolkitOutputChannel: vscode.OutputChannel,
     commandName: string,
     lambdaLocalInvokeParams: LambdaLocalInvokeParams,
+    processInvoker: SamCliProcessInvoker,
+    taskInvoker: SamCliTaskInvoker,
+    telemetryService: TelemetryService
 }): Promise<{ datum: Datum }> {
 
+    const channelLogger = getChannelLogger(toolkitOutputChannel)
     const runtime = await getRuntimeForLambda({
         handlerName: lambdaLocalInvokeParams.handlerName,
         templatePath: lambdaLocalInvokeParams.samTemplate.fsPath
     })
 
-    // TODO : Implement local run/debug in future backlog tasks
-    vscode.window.showInformationMessage(
-        `Local ${lambdaLocalInvokeParams.isDebug ? 'debug' : 'run'} support for csharp is currently not implemented.`
-    )
+    const baseBuildDir = await makeBuildDir()
+    const codeDir = path.dirname(lambdaLocalInvokeParams.document.uri.fsPath)
+    const documentUri = lambdaLocalInvokeParams.document.uri
+    const originalHandlerName = lambdaLocalInvokeParams.handlerName
+
+    const inputTemplatePath = await makeInputTemplate({
+        baseBuildDir,
+        codeDir,
+        documentUri,
+        originalHandlerName,
+        handlerName: originalHandlerName,
+        runtime: runtime,
+        workspaceUri: lambdaLocalInvokeParams.workspaceFolder.uri
+    })
+
+    const samTemplatePath: string = await executeSamBuild({
+        baseBuildDir,
+        channelLogger,
+        codeDir,
+        inputTemplatePath,
+        // TODO: add manifest path for debugger
+        manifestPath: undefined,
+        samProcessInvoker: processInvoker,
+    })
+
+    await invokeLambdaFunction({
+        baseBuildDir,
+        channelLogger,
+        configuration,
+        // TODO: Delete this
+        debugConfig: {
```

Comment:
IMO `debugConfig` should be an optional param, since it is only required for Debug.